docs/extensibility.md

```diff
@@ -1,453 +1,282 @@
-
-> **Note:** if your container registry is private, you will need to create a kubernetes secret (`kubectl create secret docker-registry crPullSecret --docker-server=<cr>  --docker-username=<cr-user> --docker-password=<cr-token>`) and access it with an `imagePullSecret`.  Here, we will assume the secret is named `crPullSecret`.
+Akri has [implemented several discovery protocols](./roadmap.md#currently-supported-protocols) with sample brokers and
+applications. However, there may be protocols you would like to use to discover resources that have not been implemented
+yet.  To enable the discovery of resources via a new protocol, you much implement a Discovery Handler (DH), which does
+discovery on behalf of the Agent. A Discovery Handler is anything that implements the `Discovery` service and
+`Registration` client defined in the [Akri's discovery gRPC proto file](../discovery-utils/proto/discovery.proto). These
+DHs run as their own Pods and are expected to register with the Agent, which hosts the `Registration` service defined in
+the gRPC interface. A discovery handler can be written in any language using protobuf; however, Akri has provided a
+template for accelerating creating a discovery handler in Rust.  
+
+This document will walk you through the development steps to implement a Discovery Handler and sample broker that
+utilizes exposed devices. This document will also cover the steps to get your Discovery Handler and broker added to
+Akri, should you wish to contribute them back.
+
+Before continuing, please read the [Akri architecture](./architecture.md), [Akri agent](./agent-in-depth.md), and
+[development](./development.md) documentation pages.  They will provide a good understanding of Akri, how it works, what
+components it is composed of, and how to build it.
+> **Note:** a Discovery Handler can use any set of steps to discover devices. It does not have to be a "protocol" in the
+> traditional sense. For example, Akri defines udev (not often called a "protocol") and OPC UA as protocols.
+
+Here, we will create a Discovery Handler to discover **HTTP-based devices** that publish random sensor data.
+
+Any Docker-compatible container registry will work for hosting the containers being used in this example (dockerhub,
+Github Container Registry, Azure Container Registry, etc).  Here, we are using the [GitHub Container
+Registry](https://github.blog/2020-09-01-introducing-github-container-registry/). You can follow the [getting started
+guide here to enable it for
+yourself](https://docs.github.com/en/free-pro-team@latest/packages/getting-started-with-github-container-registry).
```

Comment:
Link is broken.

Reply:
It seems to be working for me. Does it work for you? I ran my link checking vs code extension on it, too, and it passed.